File: mobile/src/androidTest/java/org/openhab/habdroid/TestWithoutIntro.java
package org.openhab.habdroid;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.test.InstrumentationRegistry;

import org.junit.Before;
import org.openhab.habdroid.util.Constants;

public abstract class TestWithoutIntro extends ProgressbarAwareTest {
    @Override
    @Before
    public void setup() {
        SharedPreferences.Editor edit = PreferenceManager
                .getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext())
                .edit();

        edit.putString(Constants.PREFERENCE_SITEMAP_NAME, "");
        if (preselectSitemap()) {
            edit.putString(Constants.PREFERENCE_SITEMAP_NAME, "demo");
            edit.putString(Constants.PREFERENCE_SITEMAP_LABEL, "Main Menu");
        }

        edit.putBoolean(Constants.PREFERENCE_DEMOMODE, true);
        edit.putBoolean(Constants.PREFERENCE_FIRST_START, false).commit();

        super.setup();
    }

    protected boolean preselectSitemap() {
        return false;
    }
}

File: mobile/src/androidTest/java/org/openhab/habdroid/TestWithoutIntro.kt
package org.openhab.habdroid

import androidx.core.content.edit
import androidx.test.InstrumentationRegistry

import org.junit.Before
import org.openhab.habdroid.util.Constants
import org.openhab.habdroid.util.getPrefs

abstract class TestWithoutIntro : ProgressbarAwareTest() {
    @Before
    override fun setup() {
        InstrumentationRegistry.getTargetContext().getPrefs().edit {
            putString(Constants.PREFERENCE_SITEMAP_NAME, "")
            if (preselectSitemap()) {
                putString(Constants.PREFERENCE_SITEMAP_NAME, "demo")
                putString(Constants.PREFERENCE_SITEMAP_LABEL, "Main Menu")
            }

            putBoolean(Constants.PREFERENCE_DEMOMODE, true)
            putBoolean(Constants.PREFERENCE_FIRST_START, false).commit()
        }

        super.setup()
        setupRegisterIdlingResources()
    }

    protected open fun preselectSitemap(): Boolean {
        return false
    }
}
